src/services/stats-wales-api.ts

```diff
@@ -179,14 +174,165 @@ export class StatsWalesApi {
+    }
+
+    public async getSourcesForFileImport(datasetId: string, revisionId: string, importId: string) {
+        const confirmedImport = await fetch(
+            `${this.backendUrl}/${this.lang}/dataset/${datasetId}/revision/by-id/${revisionId}/import/by-id/${importId}`,
+            {
+                method: 'GET',
+                headers: this.authHeader
+            }
+        )
+            .then((response) => {
+                if (response.ok) {
+                    return response.json();
+                }
+                const err = new HttpError(response.status);
+                err.handleMessage(response.text());
+                throw err;
+            })
+            .then((api_res) => {
+                const updatedImportDto = api_res as ImportDTO;
+                return {
+                    success: true,
+                    fileImport: updatedImportDto
+                } as ConfirmedImportDTO;
+            });
+        return confirmedImport;
+    }
+
+    public async sendCreateDimensionRequest(
+        datasetId: string,
+        revisionId: string,
+        importId: string,
+        dimensionCreationDtoArr: DimensionCreationDTO[]
+    ) {
+        const confirmedDatasetDto = await fetch(
+            `${this.backendUrl}/${this.lang}/dataset/${datasetId}/revision/by-id/${revisionId}/import/by-id/${importId}/sources`,
+            {
+                method: 'PATCH',
+                headers: { ...this.authHeader, 'Content-Type': 'application/json; charset=UTF-8' },
+                body: JSON.stringify(dimensionCreationDtoArr)
+            }
+        )
+            .then((response) => {
+                if (response.ok) {
+                    return response.json();
+                }
+                const err = new HttpError(response.status);
+                err.handleMessage(response.text());
+                throw err;
+            })
+            .then((api_res) => {
+                const datasetDTO = api_res as DatasetDTO;
+                return {
+                    success: true,
+                    dataset: datasetDTO
+                } as UploadDTO;
+            });
+        return confirmedDatasetDto;
+    }
+
+    public async uploadCSVToFixDataset(datasetId: string, revisionId: string, file: Blob, filename: string) {
+        const formData = new FormData();
+        formData.set('csv', file, filename);
+
+        const processedCSV = await fetch(
+            `${this.backendUrl}/${this.lang}/dataset/${datasetId}/revision/by-id/${revisionId}/import`,
+            {
+                method: 'POST',
+                headers: this.authHeader,
+                body: formData
+            }
+        )
+            .then((response) => {
+                if (response.ok) {
+                    return response.json();
+                }
+                const err = new HttpError(response.status);
+                err.handleMessage(response.text());
+                throw err;
```

Comment:
Lots of very similar error handling code in this class, maybe we can stick a wrapper around fetch (or use a library?) and reduce the amount of boilerplate in each function.